applications/crossbar/src/crossbar_doc.erl

```diff
@@ -246,7 +246,7 @@ patch_and_validate(Id, Context, ValidateFun) ->
     Context2 = case cb_context:resp_status(Context1) of
                    'success' ->
                        PubJObj = wh_doc:public_fields(cb_context:req_data(Context)),
-                       PatchedJObj = wh_json:merge_jobjs(PubJObj, cb_context:doc(Context1)),
+                       PatchedJObj = wh_json:merge_recursive(PubJObj, cb_context:doc(Context1)),
```

Comment:
I don't think this works like you expect, at least in my brief testing:

```erlang
1> PubJObj = {[{<<"a">>,{[{<<"b">>,<<"c">>}]}}]}.
{[{<<"a">>,{[{<<"b">>,<<"c">>}]}}]}
2> Doc = {[{<<"a">>, {[{<<"b">>, 4}]}}]}.
{[{<<"a">>, {[{<<"b">>, 4}]}}]}
3> wh_json:merge_recursive(PubJObj, Doc).
{[{<<"a">>,{[{<<"b">>,4}]}}]}
4> wh_json:merge_jobjs(PubJObj, Doc).
{[{<<"a">>,{[{<<"b">>,<<"c">>}]}}]}
```

Thoughts? What payloads are you testing this with to validate the proper and expected merging?